lib/http2.c

```diff
@@ -2105,25 +2116,28 @@ static ssize_t cf_h2_send(struct Curl_cfilter *cf, struct Curl_easy *data,
     DEBUGF(LOG_CF(data, cf, "send: nghttp2_submit_request error (%s)%u",
                   nghttp2_strerror(stream_id), stream_id));
     *err = CURLE_SEND_ERROR;
-    len = -1;
+    nwritten = -1;
     goto out;
   }
 
   infof(data, "Using Stream ID: %u (easy handle %p)",
         stream_id, (void *)data);
   stream->stream_id = stream_id;
+  /* See comment above. We assume that the whole buf was consumed by
+   * generating the request headers. */
```

Comment:
'see comment above' is too generic, maybe see TODO above or something so we don't lose track of it. not to mention that TODOs are easy to lose track of, it's better to open an issue if there is one than put it in a TODO

Reply:
I'll change to TODO. Also agree that opening an issue would be good for use of `data->state.buffer` in HTTP/2 and HTTP/3.

Reply:
Updated in recent push.